codejail_service/startup_check.py

```diff
@@ -87,13 +94,17 @@ def run_startup_safety_check():
         except BaseException as e:
             result = f"Uncaught exception from check: {e!r}"
 
+        attr_name = f"codejail.startup_check.{check['id']}"
         if result is True:
             log.info(f"Startup check {check['name']!r} passed")
+            set_custom_attribute(attr_name, 'pass')
         else:
             any_failed = True
             log.error(f"Startup check {check['name']!r} failed with: {result!r}")
+            set_custom_attribute(attr_name, 'fail')
 
     STARTUP_SAFETY_CHECK_OK = not any_failed
+    set_custom_attribute("codejail.startup_check_outcome", 'pass' if STARTUP_SAFETY_CHECK_OK else 'fail')
```

Comment:
Possibly `codejail.startup_check.status` for consistency, depending on your thoughts around the other attribute.

Reply:
I was hoping to have a distinguishable prefix between individual checks and the aggregate check to make queries easier. For example `codejail.startup_check.functionality` and `codejail.startup_check.status` share everything right up through `codejail.startup_check.*`. But maybe that distinction doesn't actually buy us anything, since (in Datadog at least) we can't do wildcards on the attribute *name* side, just the value.

Maybe `codejail.startup_check.aggregate` or `codejail.startup_check.outcome`?

Reply:
From other discussion, we'll go with `..status`.